### PR TITLE
refactor: Make all #includes relative to project root (laanwj, MeshCollider, ryanofsky)

### PR DESCRIPTION
**Pull request by meshcollider**
Rebase of #11053 

Previously started by @laanwj, ACK'ed by promag, ryanofsky, jonasschnelli, and concept-ACK'ed by practicalswift and jnewbery. Thus should be almost RTM :)

### CODE REVIEW

**Comment by MeshCollider** [comment]
@ryanofsky you might want to check this over :)

**Review by ryanofsky — COMMENTED** [review]
utACK 5d0eae757e30605ec3d26bc515856b1d20a13cc2

**Comment by laanwj** [comment]
So we're doing this?
Okay!
utACK 5d0eae7

**Comment by sipa** [comment]
Pieter was here.

**Comment by practicalswift** [comment]
utACK 5d0eae757e30605ec3d26bc515856b1d20a13cc2

Nice!

**Comment by jnewbery** [comment]
Thanks for taking and rebasing this. Big concept ACK. Will review later.

**Comment by laanwj** [comment]
Needs rebase, after that please let me know and we should just merge it, has had lots of review and is a pain to keep up to date.

File: doc/developer-notes.md [hunk 1]
```diff
@@ -461,6 +461,14 @@ namespace {
 
   - *Rationale*: Avoids confusion about the namespace context
 
+- Prefer `#include <primitives/transaction.h>` bracket syntax instead of
+  `#include "primitives/transactions.h" quote syntax when possible.
```
Comment (promag) [inline doc/developer-notes.md]: Missing `.

**Review by promag — COMMENTED** [review]
rere-utACK.

File: doc/developer-notes.md (hunk 1 above)
Comment (promag) [reply doc/developer-notes.md]: BTW, unrelated but the `Rationale` above doesn't render well. And some are missing a period.

**Comment by MeshCollider** [comment]
Rebased, will just wait for travis to verify the scripted-diff again

**Comment by jnewbery** [comment]
Tested ACK 7b91b5f8a4417da1940956c1ab32dba94dc104e1.

I noticed that there were a few `.rc` and `.mm` files still using the "" style includes. Should those also be changed to <> style? I think the change would be like this: https://github.com/jnewbery/bitcoin/commit/1a506c7ebf91747f40c9e408f8947e5491101422 but I don't have windows or mac machines to test the build on.

**Comment by theuni** [comment]
indifferent utACK 7b91b5f8a4417da1940956c1ab32dba94dc104e1.

**Comment by laanwj** [comment]
> I noticed that there were a few .rc and .mm files still using the "" style includes.

If they are correctly relative (so to include from the directory the including file is in, or even a directory relative to its directory) it's fine to keep them. `""` style includes are a problem when they're used when `<>` is meant, which was common in this project before this change.

**Pull request merged**